### PR TITLE
Reimplement the student entry layout as a responsive flexbox

### PR DESCRIPTION
**Pull request by anbenson**
Not super proud of adding 3 flexboxes + a bunch of CSS, but I think it improves on the current situation.

I did a bunch of manual testing but it might be good to do some of your own too in case there were scenarios I didn't think of (I tried most of the ones involving 2 TAs and a student).

Changes:
- add a min-height to entries so they are consistently sized for students
- move helping-text out of the button container to make it easier to deal with
- replace primary/secondary-content with three flexboxes: one overall, one around the name and question, and one around the buttons
- relayout on small viewports (hard-coded at 800px)
- buttons are no longer added dynamically but are display: none (via the .hide class) until made visible
- es6 template strings replace a bunch of string concatenations for readability. NOTE: not supported on IE.

### CODE REVIEW

File: static/css/styles.css [hunk 1]
```diff
@@ -100,21 +101,57 @@ input {
...
+    flex-shrink: 0;
+    margin: 0 5px;
+}
+
+.entry-text {
+    flex-shrink: 1;
+}
+
+.entry-name {
+    width: 100px;
```
Comment (jburns20) [inline static/css/styles.css]: I think I had this at 200px before... at 100px, it usually isn't wide enough to fit the name and Andrew ID on the same line.

File: static/css/styles.css [hunk 2]
```diff
@@ -100,21 +101,57 @@ input {
     padding: 10px 30px;
 }
 
-.primary-content {
-    float: left;
+.helping-text {
+    margin: 0 5px 5px 0;
```
Comment (jburns20) [inline static/css/styles.css]: The helping-text, when empty, is pushing everything down by 5px so the vertical centering is off. I guess we should hide and show it as necessary in addition to changing the text (or just remove the vertical margin, which is also adding unnecessary height in the student view).

Also, a side effect of moving helping-text out of the buttons container is that the name/andrewID/question text is also getting shifted down when the helping-text has content (see screenshot). Not a huge deal in my opinion and doesn't seem too easy to fix, so I think it's fine for now.

Screen Shot 2020-01-19 at 11 42 16 AM

**Comment by jburns20** [comment]
Thanks for your help with this!

File: static/css/styles.css (hunk 1 above)
Comment (anbenson) [reply static/css/styles.css]: I can increase to about 130px or so (which allows at least both of our names to fit on the same line, unlike before), but if I go longer, it pushes the buttons out of the bounding box on iPhone X. wdyt?

File: static/css/styles.css (hunk 2 above)
Comment (anbenson) [reply static/css/styles.css]: Surprisingly the :not and :empty pseudoselectors are well-supported on browsers except IE, so I just made the CSS conditional on .helping-text being non-empty.

For your latter comment - I did that intentionally. The container height is going to increase regardless, so I thought it best to keep the alignment of text with buttons consistent.

File: static/css/styles.css (hunk 1 above)
Comment (jburns20) [reply static/css/styles.css]: Can we do 200px and then shorten it to 130 below 800 width?

File: static/css/styles.css (hunk 2 above)
Comment (jburns20) [reply static/css/styles.css]: Okay, sounds good

File: static/css/styles.css (hunk 1 above)
Comment (anbenson) [reply static/css/styles.css]: Yep, done.

**Pull request merged**